libvirt/driver_libvirt.go

```diff
@@ -180,3 +185,53 @@ func (d *DomainImpl) GetInfo() (res DomainInfo, err error) {
 	}
 	return
 }
+
+func (d *DomainImpl) GetVolumeInfo() (res []VolumeInfo, err error) {
+	if volumeInfoStr, err := d.domain.QemuMonitorCommand(qemuMonitorCommand, domainQemuMonitorCommandFlags); err == nil {
+		res = d.parseVolumeInfo(volumeInfoStr)
+	}
+	return
+}
+
+func (d *DomainImpl) parseVolumeInfo(volumeInfoStr string) []VolumeInfo {
+	var result []VolumeInfo
+	split := strings.Split(volumeInfoStr, "\n")
+	r, _ := regexp.Compile("json:{(.*)}")
+	for _, line := range split {
+		if match := r.FindString(line); match != "" {
+			var objmap1 map[string]json.RawMessage
+			var objmap2 map[string]string
+			b := []byte(match[5:]) // match without the "json:" prefix
+			if err := json.Unmarshal(b, &objmap1); err == nil {
+				if err := json.Unmarshal(objmap1["file"], &objmap2); err == nil {
+					pool := ""
```

Comment:
These four lines are equivalent to pool := objmap2["pool"]
When there is no map entry, the null value (empty string) is returned.